immunopepper/mode_cancerspecif.py

```diff
@@ -140,15 +140,27 @@ def mode_cancerspecif(arg):
                 = check_interm_files(cancer_out, arg.cohort_expr_support_cancer, arg.n_samples_lim_cancer, \
                                      target_sample=cancer_sample, tag=f'cancer_{mutation_mode}', batch_tag=batch_tag)
 
-            ## Cancer file is matrix
-            if arg.path_cancer_matrix_segm or arg.path_cancer_matrix_edge:
+            ## Cancer file checks
+            if arg.path_cancer_matrix_segm or arg.path_cancer_matrix_edge or arg.paths_cancer_samples:
                 logging.info("\n \n >>>>>>>> Preprocessing Cancer sample {}  ".format(cancer_sample_ori))
                 mutation_mode = arg.mut_cancer_samples[0]
+                if arg.paths_cancer_samples:
+                    try:
+                        path_kmer_file = [arg.paths_cancer_samples[cix]]
+                    except:
+                        logging.error(f'--ids_cancer_samples not matching --paths_cancer_samples, exit.')
```

Comment:
Th above `try` does not check for any matching. It only tries to access an array at an index, no?

Reply:
It does some matching for length: 

Consider --ids_cancer_samples=['sample_A', 'sample_B'] (named to be used for save) 
and --paths_cancer_samples=['path_sample_A', 'path_sample_B']
Full documentation below
>     helpers.add_argument("--ids-cancer-samples", nargs='+',help=" list of all cancer samples on which to apply the filtering. If --paths-cancer-samples provided they should be given in same order", required=True, default='')

>     crf.add_argument("--paths-cancer-samples", nargs='+', help="List of single cancer sample files", required=False, default='')

then 
```
cix, cancer_sample_ori in enumerate(arg.ids_cancer_samples):
[...]
try:
                        path_kmer_file = [arg.paths_cancer_samples[cix]]
                    except:
                        logging.error(f'--ids_cancer_samples not matching --paths_cancer_samples, ex

```
Will just check that the user did not forget to enter two lists of the same length. Whether there is an actual matching is not something the software can check.